fakeredis/_server.py

```diff
@@ -898,11 +908,18 @@ def _scan(self, keys, cursor, *args):
         result_cursor = cursor + count
         result_data = []
 
-        if pattern is not None:
-            regex = compile_pattern(pattern)
+        regex = compile_pattern(pattern) if pattern is not None else None
+
+        def _match_key(key):
```

Comment:
No need for the leading underscore - the name isn't accessible from outside the method.

Reply:
Fixed.